src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -326,6 +327,141 @@ def promote(self, name):
             raise CallError(f'Failed to promote dataset: {e}')
 
 
+class ZFSSnapshotTask(CRUDService):
```

Comment:
zfs.* namespace is being reserved for private services.

Since snapshot task is a user facing service I suggested this to be moved to its own plugin or some related.